### DISCUSSION TITLE
The data source for the AI models (Open AI Vs. Azure OpenAI)

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [microsoft/teams-sdk#183] [Q&A] [answered]
The diagram in the read me shows:
![image](https://github.com/microsoft/teams-ai/assets/5054931/a23596ed-7fe4-457b-8450-e568152fae3d)

but the samples, the "Getting started" and the "04.ai.a.teamsChefbot" all refers to getting an API key from Open AI which has models based on public internet data.

How do we build an application based on data within the 365 tenant? 
How does the library connects to graph?

**Comment by corinagum** [comment, marked as answer]
Hi @Ofer-Gal, my apologies for the late reply. 

Regarding Azure OpenAI: please see the [sample request](https://github.com/microsoft/teams-ai/issues/214) issue that is requesting the same information. I will post an example there and look into getting a project into the samples folder. 

Regarding your second and third questions, when you write a bot, the bot is created on the same tenant as the 365 data. Implementing this with graph is not necessarily related to AI or Teams. Have you taken a look at the existing [BotBuilder graph sample](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/javascript_nodejs/24.bot-authentication-msgraph/README.md) and the [Microsoft learning course?](https://learn.microsoft.com/en-us/samples/microsoftgraph/msgraph-sample-botframework/microsoft-graph-bot-framework-sample/)

Like a non-AI bot, you would then be able to access tenant/graph information via authentication and create handlers to code your solution with or without Teams features or AI.

**Reply by Ofer-Gal** [reply]
Thanks

**Comment by Ofer-Gal** [comment]
Good answer
